Model/Payment/Customer.php

```diff
@@ -46,28 +83,32 @@ public function findOrCreate(Order $order)
     {
         $billing = $order->getBillingAddress();
         $customer = null;
-        $customerId = null;
+        $vindiCustomerId = null;
 
         if (!$order->getCustomerIsGuest()) {
             $customer = $this->customerRepository->get($billing->getEmail());
-            $customerId = $this->findVindiCustomer($customer->getId());
+            $vindiCustomerId = $this->findVindiCustomerIdByCustomerId($customer->getId());
         }
 
-        if ($customerId) {
-            if($order->getPayment()->getMethod() == "vindi_pix") {
+        if ($vindiCustomerId) {
+            if ($order->getPayment()->getMethod() == "vindi_pix") {
                 $customerVindi = $this->getVindiCustomerData($customer->getId());
```

Comment:
Uma dúvida. Como alteramos o code do cliente para uma `url base + id + timestamp` ou `url base + email + timestamp`. Essa query possivelmente retornará vazia. Isto impacta na atualização do documento em comprar como convidado, certo?

Reply:
> Uma dúvida. Como alteramos o code do cliente para uma `url base + id + timestamp` ou `url base + email + timestamp`, essa query possivelmente retornará vazia. Isto impacta na atualização do documento em compras como convidado, certo?

Essa alteração, utilizando o email na composição do 'code', seria aplicada apenas para compras de convidados. 
Como o Magento não registra clientes para essas compras, não teríamos o 'customer_id' disponível. 
Além disso, uma outra solução possível nesse fluxo em específico seria criar um customer na Vindi sem o campo 'code', visto que o mesmo é um campo é opcional.

Reply:
Esta função `getVindiCustomerData` busca o cliente pelo code. No caso, a chamada `$customer->getId()` retornaria nulo? Caso positivo, entendo que o código dentro da condicional `$order->getPayment()->getMethod() == "vindi_pix"` não irá funcionar como o esperado visto que o método `getVindiCustomerData` nunca irá retornar o cliente devido a inexistência do code na Vindi.
![image](https://github.com/user-attachments/assets/beb3d4af-4b0e-463c-9197-1cf59209f4e2)

Reply:
Antes de chamar o `getVindiCustomerData` há uma verificação se `$vindiCustomerId` existe. E ele só irá existir quando o cliente não for convidado `!$order->getCustomerIsGuest()`.
Por isso, nesse fluxo de compra como convidado, o método `getVindiCustomerData` não será executado.